### ISSUE TITLE
Podman API Service fails to start

### ISSUE THREAD

**Issue by aidalgol** [NixOS/nixpkgs#194155]
### Describe the bug
The podman systemd user service fails to start with the error `Failed to locate executable podman: No such file or directory`

### Steps To Reproduce
1. Set the NixOS config option `virtualisation.podman.enable = true;`
2. Rebuild
3. Log in to a normal user account and run `systemctl --user start podman`
4. Run `systemctl --user status podman`

### Expected behavior
The podman API service to start successfully.

### Additional context
I am running a ZFS system, and have added `virtualisation.podman.extraPackages = [ pkgs.zfs ];` to my system config.

My user graphical login is via `sx`, and my user profile is using a flake-based `home-manager` config.

### Notify maintainers
@NixOS/podman 

### Metadata
(Note: I am using flake-based configurations for system and user profiles, so the channels below are not in use.)
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.70, NixOS, 22.05 (Quokka), 22.05.20220929.13cbe53`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(aidan): `"nixpkgs"`
 - channels(root): `"nixos"`
 - nixpkgs: `/home/aidan/.nix-defexpr/channels/nixpkgs`
```

**Comment by zowoq** [comment]
Is is reproducible on unstable?

**Comment by aidalgol** [comment]
I don't know, and I don't have an unstable system on which I could test.

**Comment by yurkobb** [comment]
Tested with `podman` from unstable (`4.2.1`) by adding this to my otherwise 22.05 `configuration.nix`:

```nix
nixpkgs.config.packageOverrides = pkgs: {
  podman = unstable.podman;
};
```

Getting the same error.

I'm guessing the problem is that the systemd unit (`/etc/systemd/user/podman.service`) doesn't contain the absolute path to the podman executable:

```
ExecStart=podman $LOGGING system service
```

As a workaround, I added the following to `/home/$USER/.config/systemd/user/podman.service.d/override.conf`:
```ini
[Service]
ExecStart=
ExecStart=/run/current-system/sw/bin/podman $LOGGING system service
```

After running `systemctl --user daemon-reload` and `systemctl --user restart podman.socket` the service starts.

Perhaps the systemd unit needs to be patched to contain the nix store path to podman?
